bql/semantic/expression.go

```diff
@@ -197,6 +153,33 @@ func (e *comparisonForLiteral) Evaluate(r table.Row) (bool, error) {
 	}
 }
 
+// comparisonForNodeLiteral represents the internal representation of an expression of comparison between a binding and a node literal.
+type comparisonForNodeLiteral struct {
+	op OP
+
+	lB  string
+	rNL string
+}
+
+func (e *comparisonForNodeLiteral) Evaluate(r table.Row) (bool, error) {
+	eL, err := getValue(e.lB, r)
+	if err != nil {
+		return false, err
+	}
+
+	csEL, csER := formatCell(eL), strings.TrimSpace(e.rNL)
+	switch e.op {
+	case EQ:
+		return reflect.DeepEqual(csEL, csER), nil
+	case LT:
+		return csEL < csER, nil
+	case GT:
```

Comment:
I am beginning to wonder if we should have checked if the type of the binding is correct and return error if not.
WDYT roger?

Reply:
I believe it makes sense too, I am adding these checks right now.